octomap/include/octomap/OcTreeBaseImpl.h

```diff
@@ -108,6 +111,57 @@ namespace octomap {
     inline unsigned int getTreeDepth () const { return tree_depth; }
 
     inline double getNodeSize(unsigned depth) const {assert(depth <= tree_depth); return sizeLookupTable[depth];}
+    
+    
+    // -- Tree structure operations formerly contained in the nodes ---
+   
+    /// Creates (allocates) the i-th child of the node. @return ptr to newly create NODE
+    NODE* createNodeChild(NODE* node, unsigned int childIdx);
```

Comment:
Yeah, it gets philosophical. Question is if one wants to make the distinction between the logical tree (this is changed) and the BaseImpl class (that remains unchanged, right?). On the same level one could also argue to make the methods static since they don't touch/read any members of the BaseImpl instance... But I agree that from the outside this is all very confusing.